### ISSUE TITLE
Error for async hasMany and store.unloadAll

### ISSUE THREAD

**Issue by shamcode** [emberjs/data#5063]
Hi!
I have two models:
```js
//models/foo.js
export default Model.extend({
  bars: hasMany('bar', {async: true})
});

//models/bar
export default DS.Model.extend({});
```
Sample response for `store.findAll('foo')` (pay attention to **links**):
```json
{
  "data": {
   "id": 1,
   "type": "foo",
   "relationships": {
     "bars": {
       "links": { "related": "bars" }
     }
   }
}
```
And then i try load relationships with code:
```js
// routes/application.js
export default Ember.Route.extend({
  model() {
    return this.store
      .findRecord('foo', 1).then((foo) => foo.get('bars'))
      .then(() => {
        this.store.unloadAll('foo'); // If comment this row, tests pass
        return this.store.findRecord('foo', 1).then((foo) => foo.get('bars'))
      });
  }
});
```
Ember throw error
```
Error while processing route: index Assertion Failed: calling set on destroyed object: <DS.PromiseManyArray:ember257>.content = <DS.ManyArray:ember258> Error
    at Object.assert (http://localhost:4200/assets/vendor.js:24286:15)
    at set (http://localhost:4200/assets/vendor.js:35785:45)
    at Class.set (http://localhost:4200/assets/vendor.js:49787:34)
    at ManyRelationship._updateLoadingPromise (http://localhost:4200/assets/vendor.js:161232:31)
    at ManyRelationship.getRecords (http://localhost:4200/assets/vendor.js:161451:19)
    at Class.get (http://localhost:4200/assets/vendor.js:169637:58)
    at ComputedPropertyPrototype.get (http://localhost:4200/assets/vendor.js:36390:28)
    at get (http://localhost:4200/assets/vendor.js:35698:19)
    at Class.get (http://localhost:4200/assets/vendor.js:49721:34)
    at http://localhost:4200/assets/ember-data-async-has-many-after-unload-bug.js:394:22
```
[Repo for reproduce (just run acceptance test)](https://github.com/shamcode/ember-data-load-async-has-many-after-unload/tree/master)
Collected ember-try report:
```
------ RESULTS ------

Scenario ember-data 2.15.0-beta.2: FAIL
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-data         │ 2.15.0-beta.2      │ 2.15.0-beta.2                │ npm      │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-data 2.14.4: FAIL
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-data         │ 2.14.4             │ 2.14.4                       │ npm      │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-data 2.13.2: FAIL
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-data         │ 2.13.2             │ 2.13.2                       │ npm      │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘

Scenario ember-data 2.12.2: SUCCESS
Command run: ember test
┌────────────────────┬────────────────────┬──────────────────────────────┬──────────┐
│ Dependency         │ Expected           │ Used                         │ Type     │
├────────────────────┼────────────────────┼──────────────────────────────┼──────────┤
│ ember-data         │ 2.12.2             │ 2.12.2                       │ npm      │
└────────────────────┴────────────────────┴──────────────────────────────┴──────────┘


3 scenarios failed
1 scenarios succeeded
4 scenarios run
```
If i don't use **links** for `hasMany`, all work perfectly: [ember-twiddle](https://ember-twiddle.com/42f57ee1392eea3604b265016d915922?openFiles=models.bar.js%2C)

**Comment by stefanpenner** [comment]
Thanks. I think this is related to some of the recent model resurrection stuff that was recently added.  We have been working to tidy it up, I will make sure this case is added to the scenarios that are problematic.

If you have some cycles, submitting a pill request with a failing test would be very helpful.

**Comment by shamcode** [comment]
@stefanpenner I've add tests for this behavior, but I think they aren't clear. Test contains override for method _ajaxRequest with setTimeout. I think, jQuery use this method for AJAX
Should I fix this and do not use setTimeout?

**Comment by shamcode** [comment]
Related  issue #4986

**Comment by runspired** [comment]
I believe this is resolved as of #5378. Please feel free to re-open if you find this is still a bug on the latest master.
